### PR TITLE
Add API for postMessage with PostMessageOptions

### PR DESCRIPTION
**Pull request by dtapuska**
This is similar to the HTML changes in https://github.com/whatwg/html/pull/3800

***
<a href="https://pr-preview.s3.amazonaws.com/dtapuska/ServiceWorker/pull/1344.html" title="Last updated on Mar 7, 2019, 2:25 PM UTC (999e1bd)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/ServiceWorker/1344/b4a9916...dtapuska:999e1bd.html" title="Last updated on Mar 7, 2019, 2:25 PM UTC (999e1bd)">Diff</a>

### CODE REVIEW

**Comment by domenic** [comment]
The corresponding HTML changes have been merged.

File: docs/index.bs [hunk 1]
```diff
@@ -309,6 +309,7 @@ spec: webappsec-referrer-policy; urlPrefix: https://w3c.github.io/webappsec-refe
         readonly attribute USVString scriptURL;
         readonly attribute ServiceWorkerState state;
         void postMessage(any message, optional sequence&lt;object&gt; transfer = []);
+        void postMessage(any message, PostMessageOptions options);
```
Comment (annevk) [inline docs/index.bs]: This should be marked optional, the previous line should have its second argument marked as non-optional (and non-defaulting).

**Comment by dtapuska** [comment]
@annevk changes made

**Comment by mattto** [comment]
Hi @dtapuska sorry that this never got merged. Can you rebase and reupload? You don't need the docs/index.html change, that gets generated automatically after you commit.

I could attempt to do this manually but I'm afraid the commit would lose attribution to you.

**Comment by annevk** [comment]
@mattto it shouldn't lose attribution. If it does, you can override the author explicitly using `--author`.

**Comment by dtapuska** [comment]
Rebased and forced pushed

**Comment by mattto** [comment]
Thanks all, I ended up doing the manual merge to drop the change to index.html. It landed in https://github.com/w3c/ServiceWorker/commit/3b3197431df710b2d60a69cdb94be6a63cc64f6d and I expect index.html to be generated automatically soon.

**Pull request closed without merging**